components/dashboard/src/settings/Preferences.tsx

```diff
@@ -49,13 +50,30 @@ export default function Preferences() {
         }
     };
 
+    const setWorkspaceClass = async (value: string) => {
+        const additionalData = user?.additionalData || {};
+        const prevWorkspaceClass = additionalData?.workspaceClasses?.regular;
+        const workspaceClasses = (additionalData?.workspaceClasses || {}) as WorkspaceClasses;
+        workspaceClasses.regular = value;
+        workspaceClasses.prebuild = value;
+        additionalData.workspaceClasses = workspaceClasses;
+        if (value !== prevWorkspaceClass) {
+            await getGitpodService().server.updateLoggedInUser({ additionalData });
+        }
+        return prevWorkspaceClass;
+    };
+
     return (
         <div>
             <PageWithSettingsSubMenu title="Preferences" subtitle="Configure user preferences.">
                 <h3>Editor</h3>
                 <p className="text-base text-gray-500 dark:text-gray-400">Choose the editor for opening workspaces.</p>
                 <SelectIDE location="preferences" />
-                <SelectWorkspaceClass enabled={BillingMode.canSetWorkspaceClass(userBillingMode)} />
+                <SelectWorkspaceClass
+                    workspaceClass={user?.additionalData?.workspaceClasses?.regular}
+                    enabled={BillingMode.canSetWorkspaceClass(userBillingMode)}
+                    setWorkspaceClass={setWorkspaceClass}
+                />
```

Comment:
**suggestion**: Once this is merged, it could be worth updating the existing and soon [outdated docs](https://www.gitpod.io/docs/configure/workspaces/workspace-classes#:~:text=You%20can%20configure%20the%20workspace%20class%20that%20should%20be%20used%20for%20your%20workspaces%20in%20your%20user%20preferences.).

> You can configure the workspace class that should be used for your workspaces in your user preferences.